markdown_lab/core/config.py

```
@@ -0,0 +1,200 @@
+            "MARKDOWN_LAB_CACHE_DIR": ("cache_dir", str),
+            "MARKDOWN_LAB_PARALLEL_WORKERS": ("parallel_workers", int),
+            "MARKDOWN_LAB_RUST_BACKEND": ("rust_backend_enabled", lambda x: x.lower() == "true"),
+        }
+
+        for env_var, (attr_name, type_converter) in env_mappings.items():
+            env_value = os.getenv(env_var)
+            if env_value is not None:
+                try:
+                    setattr(self, attr_name, type_converter(env_value))
+                except (ValueError, TypeError) as e:
+                    raise ValueError(f"Invalid environment variable {env_var}={env_value}: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)